core/senders/qq_send_prep.py

```
@@ -0,0 +1,156 @@
+FlattenableBatches: TypeAlias = Sequence[Batch | NestedBatches]
+
+
+def _as_str_object_mapping(value: object) -> Mapping[str, object]:
+    """当值可视为映射时返回映射视图，否则返回空映射。"""
+    if isinstance(value, Mapping):
+        return value
+    return {}
+
+
+def _is_nested_batches(item: Batch | NestedBatches) -> TypeGuard[NestedBatches]:
+    """判断是否为一层嵌套的批次分组，例如 ``[[msg1], [msg2]]``。"""
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)